openmp/libomptarget/plugins-nextgen/common/PluginInterface/PluginInterface.cpp

```diff
@@ -274,7 +317,7 @@ struct RecordReplayTy {
   void saveKernelOutputInfo(const char *Name) {
     SmallString<128> OutputFilename = {
         Name, (isRecording() ? ".original.output" : ".replay.output")};
-    dumpDeviceMemory(OutputFilename);
+    dumpDeviceMemory(OutputFilename, /*saveDiff*/ true);
```

Comment:
If `saveDiff` is always `true` when replaying then this is incompatible with verification in the `llvm-omp-kernel-replay` tool, right? We should set `saveDiff` either through an env var or through the `llvm-omp-kernel-replay` tool (latter is better) and make the `llvm-omp-kernel-replay` tool aware. If `saveDiff` is true then the tool should just show the differences already saved in the diff files.

Reply:
saveDiff isn't always true. For pre-record memory dump, saveDiff is false (on line 302)

Reply:
OK, I understand better now. You basically save the diff from input memory for kernel outputs on both record and replay. This make more sense to me and it is indeed compatible with verification in the replay tool.

Reply:
Create a different function `dumpDeviceMemoryDiff` and put the logic there, instead of conflating the existing one with the `saveDiff` flag